src/aktualizr_lite/helpers.cc

```diff
@@ -88,21 +106,22 @@ LiteClient::LiteClient(Config &config_in) : config(std::move(config_in)), primar
   headers.push_back(header);
   add_apps_header(headers, config.pacman);
 
-  std::pair<bool, Uptane::Target> pair = finalizeIfNeeded(*storage, config.pacman);
-  headers.emplace_back("x-ats-target: " + pair.second.filename());
-  headers.emplace_back("x-ats-tags: " + boost::algorithm::join(config.pacman.tags, ","));
-
   http_client = std::make_shared<HttpClient>(&headers);
   report_queue = std_::make_unique<ReportQueue>(config, http_client);
+  package_manager = PackageManagerFactory::makePackageManager(config.pacman, config.bootloader, storage, http_client);
 
-  if (pair.first) {
-    notifyInstallFinished(pair.second, data::ResultCode::Numeric::kOk);
-  }
+  primary = std::make_shared<SotaUptaneClient>(config, storage, http_client);
+
+  std::pair<Uptane::Target, data::ResultCode::Numeric> pair = finalizeIfNeeded(*package_manager, *storage, config.pacman);
+  headers.emplace_back("x-ats-target: " + pair.first.filename());
+  headers.emplace_back("x-ats-tags: " + boost::algorithm::join(config.pacman.tags, ","));
```

Comment:
Hm, what I was trying here is to run finalizeIfNeeded after setBootOK, which gets called via makePackageManager, so we can have the right call order.

As we need the headers before instantiating httpclient (and pair.first.filename), I guess there is no easy way to avoid finalizing before setting boot ok, but I guess this should be fine still.

Reply:
I think the httpclient now has an "update-header" type API that i got upstream. So you could use that if it makes things cleaner.